src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +111,10 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(c -> candidateValidator.test(c))
```

Comment:
it's enough to pass object of predicate into filter operation

```suggestion
                .filter(candidateValidator)
```